### ISSUE TITLE
The library should have at least one runtime dependency to be really helpful

### ISSUE THREAD

**Issue by ClausKlein** [bemanproject/exemplar#59]
An STATIC, SHARED buildable library should be created, not only a simple header only (INTERFACE).

It should also export only it public interface!

See too https://crascit.com/2019/10/16/cppcon-2019-deep-cmake-for-library-authors/

**Comment by camio** [comment]
This would be great. @ClausKlein would you be willing to contribute something like this?

**Comment by ClausKlein** [comment]
Hhure, I would like to help.

**Comment by camio** [comment]
Thanks! Feel free to create a PR or throw some ideas here. Let me know if there's anything I can do to help.

**Comment by ClausKlein** [comment]
Do you now [WindowsToolchain](https://github.com/MarkSchofield/WindowsToolchain)?
I would like to use them for windows CI build with `ninja`.

This builds quite faster and makes it even more easy to setup a `cmake workflow preset` for windows builds.

**Comment by bretbrownjr** [comment]
Why does this project need a runtime dependency? I think I need to hear some more concrete situations where runtime dependencies are required or at least clearly superior to other options.

Note that this project is intentionally not an educational project to teach CMake [1]. It is intended to assist a reasonably experienced C++ engineer to start developing a standard library proposal.

I would expect the typical proposal would have a fairly modest dependency list, frequently an empty one, but maybe I am too optimistic.

[1] Jason Turner's https://github.com/cpp-best-practices/cmake_template is more in line for those goals.

**Comment by camio** [comment]
> Why does this project need a runtime dependency? I think I need to hear some more concrete situations where runtime dependencies are required or at least clearly superior to other options.

This is a fair question. I would imagine something the linear algebra proposal would have a runtime dependency on blas. The, now defunct, 2d graphics proposal would have a runtime dependency on cairo IIRC.

**Comment by camio** [comment]
> Do you know the [WindowsToolchain](https://github.com/MarkSchofield/WindowsToolchain)?

I was unfamiliar with that. Thanks for pointing it out!

My understanding is that most professional Windows developers use Visual Studio, but I could be out-of-date.

**Comment by ClausKlein** [comment]
> My understanding is that most professional Windows developers use Visual Studio, but I could be out-of-date.

Yes and No, many I know use also `qt-creator, clion`, but not on **CI**.

This toolchain also supports [clang-cl](https://github.com/aminya/project_options/tree/main?tab=readme-ov-file#library-usage) with `ninja`. (`clang-19` is included in `Visual Studio 2023`, and `cmake workflows` are supported too)

**Comment by camio** [comment]
I took a look at the [ISO Cpp 2024 developer survey](https://isocpp.org/files/papers/CppDevSurvey-2024-summary.pdf) and this is the breakdown for primary IDE/Editor used:

![image](https://github.com/user-attachments/assets/4fdd93c1-354d-42f6-b7b4-20f2b9010001)

**Comment by bretbrownjr** [comment]
>  I would imagine something the linear algebra proposal would have a runtime dependency on blas. The, now defunct, 2d graphics proposal would have a runtime dependency on cairo IIRC.

Sure, but *this* library doesn't need a runtime dependency. I would think less is more when it comes to getting started on a new library.

Of course, I would support someone documenting the relevant steps to add a dynamic dependency. Or better yet, someone working on a proposal that benefits from a dynamic dependency.

**Comment by wusatosi** [comment]
Maybe a `fmt` vcpkg dependency for testing just because?

I actually don't really think we need package manager, besides Google Test, all current projects are dependency-free.

**Comment by ClausKlein** [comment]
See https://github.com/beman-project/net29/blob/cb731f8bda0d964011eab0cb0c5e21ee2bbd7928/CMakeLists.txt#L17
